gh-103109: Add documentation for `ignore_warnings` (#103110)
python/cpython: Doc/library/test.rst

.. versionadded:: 3.10

.. function:: ignore_warnings(*, category)

   A decorator to suppress warnings. *category* is a subclass of :exc:`Warning`
   class. It is approximately equivalent to calling :func:`warnings.catch_warnings()`
   with :meth:`warnings.simplefilter` set to ``ignore``. If you want to suppress some

>    known warnings in test case, using code like the following::

Reviewer: ```suggestion
   specific known warnings in a test case, use code like the following::
```

Fix several grammar errors and phrasing issues

      @warning_helper.ignore_warnings(category=DeprecationWarning)
      def test_suppress_warning():

>           # do something

Reviewer: ```suggestion
          # do something

   .. versionadded:: 3.8          
```

Add versionadded per @erlend-aasland 's comment (sorry, though it was added in 3.10 like the rest, but in fact the notation above was due to them being moved to a new submodule instead).

.. versionadded:: 3.10

.. function:: ignore_warnings(*, category)

   A decorator to suppress warnings. *category* is a subclass of

>    :exc:`Warning` class. It is approximately equivalent to calling
>    :func:`warnings.catch_warnings()` with :meth:`warnings.simplefilter`
>    set to ``ignore``. If you want to suppress some specific known warnings
>    in a test case, use code like the following::

Reviewer 2: Sorry to pick on this further, but I think this may be too verbose for the `test.support` docs, as they are primarily written for CPython core devs and contributors.

I'd prefer a simpler variant, kind of like this:

```suggestion
   :exc:`Warning`.
   Example use::
```

Alternative:

```suggestion
   the :exc:`Warning` class.
   Example use::
```

@CAM-Gerlach, what do you think?

Reviewer: It could definitely say the same thing in a lot fewer words, though considering this whole page is specifically for internal use, I wouldn't cut out _that_ many relevant details.—in particular, the  Also, looking at it more, I spotted some other things that can be fixed/improved. I've made another suggestion covering the full block that does both (and uses SemBr).

Reviewer: Looks like the agreed suggestion was applied, so marking this as resolved, thanks.

>    A decorator to suppress warnings. *category* is a subclass of
>    :exc:`Warning` class. It is approximately equivalent to calling
>    :func:`warnings.catch_warnings()` with :meth:`warnings.simplefilter`
>    set to ``ignore``. If you want to suppress some specific known warnings
>    in a test case, use code like the following::

Reviewer: ```suggestion
   Suppress warnings that are instances of *category*,
   which must be :exc:`Warning` or a subclass.
   Roughly equivalent to :func:`warnings.catch_warnings`
   with :meth:`warnings.simplefilter('ignore', category=category) <warnings.simplefilter>`.
   For example::
```

* Rewrite more clearly and concisely
* Clarify explicit what *category* does
* Avoid implying that *category* must be a _strict_ subclass of `Warning`
* State what it is equivalent to more explicitly.
* Fix some minor grammar/phrasing issues I missed the first time

Reviewer 2: Looks good to me. @CharlieZhao95 can you apply CAM's suggestion?

Author: > Looks good to me. @CharlieZhao95 can you apply CAM's suggestion?

Sure. Thanks to everyone for the careful review and suggestions 😄

Reviewer: Thanks for your contribution!
